tutorials/llm/llama-3/nemo2-sft-peft/nemo2-peft.ipynb

```diff
@@ -0,0 +1,540 @@
+    "    )\n",
+    "    trainer = run.Config(\n",
+    "        nl.Trainer,\n",
+    "        devices=1,\n",
+    "        max_steps=20,\n",
+    "        accelerator=\"gpu\",\n",
+    "        strategy=strategy,\n",
+    "        plugins=bf16_mixed(),\n",
+    "        log_every_n_steps=1,\n",
+    "        limit_val_batches=2,\n",
+    "        val_check_interval=2,\n",
+    "        num_sanity_val_steps=0,\n",
+    "    )\n",
+    "    return trainer\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Logger\n",
+    "Configure your training steps, output directories and logging through `NeMoLogger`. In the following example, the experiment output will be saved at `./results/nemo2_peft`.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def logger() -> run.Config[nl.NeMoLogger]:\n",
+    "    ckpt = run.Config(\n",
+    "        nl.ModelCheckpoint,\n",
+    "        save_last=True,\n",
+    "        every_n_train_steps=10,\n",
+    "        monitor=\"reduced_train_loss\",\n",
+    "        save_top_k=1,\n",
+    "        save_on_train_epoch_end=True,\n",
+    "        save_optim_on_train_end=True,\n",
+    "    )\n",
+    "\n",
+    "    return run.Config(\n",
+    "        nl.NeMoLogger,\n",
+    "        name=\"nemo2_peft\",\n",
+    "        log_dir=\"./results\",\n",
+    "        use_datetime_version=False,\n",
+    "        ckpt=ckpt,\n",
+    "        wandb=None\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "\n",
+    "### Optimizer\n",
+    "In the following example, we will be using distributed adam optimizer, and pass in optimizer configuration through `OptimizerConfig`: \n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def adam_with_cosine_annealing() -> run.Config[nl.OptimizerModule]:\n",
+    "    opt_cfg = run.Config(\n",
+    "        OptimizerConfig,\n",
+    "        optimizer=\"adam\",\n",
+    "        lr=0.0001,\n",
+    "        adam_beta2=0.98,\n",
+    "        use_distributed_optimizer=True,\n",
+    "        clip_grad=1.0,\n",
+    "        bf16=True,\n",
+    "    )\n",
+    "    return run.Config(\n",
+    "        nl.MegatronOptimizerModule,\n",
+    "        config=opt_cfg\n",
+    "    )\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### LoRA Adapter\n",
+    "We need to pass in LoRA adapter to our finetuning API to perform LoRA finetuning. We can configure adapter like the following. The target module we support includes: `linear_qkv`, `linear_proj`, `linear_fc1` and `linear_fc2`. In the final script we used default configurations for LoRA (`llm.peft.LoRA()`), which will use the full list with `dim=32`."
```

Comment:
fix punctuation, add definitive article "the", and revise sentence

    "We need to pass in the LoRA adapter to our fine-tuning API to perform LoRA fine-tuning. We can configure the adapter as follows. The target module we support includes: `linear_qkv`, `linear_proj`, `linear_fc1` and `linear_fc2`. In the final script, we used the default configurations for LoRA (`llm.peft.LoRA()`), which will use the full list with `dim=32`."